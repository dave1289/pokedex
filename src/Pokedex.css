.Pokedex-title {
   text-align: center;
   display: block;
}

.Pokedex {
   width: 50vw;
   position:relative;
   margin:auto;
   display:flex;
   align-items:center;
   justify-content:center;
   flex-flow:row wrap;
}