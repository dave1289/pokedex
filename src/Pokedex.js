import React from 'react'
import Pokecard from './Pokecard'
import './Pokedex.css'

const Pokedex = ({ pokemonArray }) => {
   return (
      <>
         <h1 className='Pokedex-title' id='1'>Pokedex</h1>
         <div className='Pokedex'>
            {pokemonArray.map(p => (
               <div className='Pokedex-Pokecards'>
                  <Pokecard id={p.id} name={p.name} type={p.type} bxp={p.base_experience} />
               </div>))}
         </div>
      </>
   )
}

export default Pokedex