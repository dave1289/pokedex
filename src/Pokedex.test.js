import { render, screen } from '@testing-library/react';
import React from "react"
import Pokedex from './Pokedex';


it('renders pokedex', () => {
   render(<Pokedex pokemonArray={[
      {id: 149, name: 'Dragonite', type: 'dragon-flying', base_experience: 300}
    ]}/>);
});

it('matches snapshot', () => {
   const { asFragment } = render(<Pokedex pokemonArray={[
      {id: 149, name: 'Dragonite', type: 'dragon-flying', base_experience: 300}]} />)
   expect(asFragment()).toMatchSnapshot()
})