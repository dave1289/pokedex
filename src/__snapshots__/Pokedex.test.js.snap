// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="Pokedex-title"
    id="1"
  >
    Pokedex
  </h1>
  <div
    class="Pokedex"
  >
    <div
      class="Pokedex-Pokecards"
    >
      <div
        class="Pokecard"
        id="149"
      >
        <h1
          class="Pokecard-name"
        >
          Dragonite
        </h1>
        <br />
        <img
          alt="Dragonite"
          class="Pokecard-img"
          src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/149.png"
        />
        <br />
        <sub
          class="Pokecard-type"
        >
          Type: dragon-flying
        </sub>
        <br />
        <sub
          class="Pokecard-xp"
        >
          EXP: 300
        </sub>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
